### PR TITLE
Added 2 new Endpoints and verified listGuildMembers

### PR DESCRIPTION
**Pull request by noahknegt**
Implemented 2 new endpoints, the modifyCurrentUserNick and searchGuildMembers. And i verified the working of the listGuildMembers endpoint.

### CODE REVIEW

File: discordpp/categories/guild.hh [hunk 1]
```diff
@@ -464,8 +464,7 @@ sptr<const std::string> render_target() override {
     }
     return std::make_shared<const std::string>(out);
 }
-#include <discordpp/macros/defineCallClose.hh>*/
-
+#include <discordpp/macros/defineCallClose.hh>
 
 //https://discord.com/developers/docs/resources/guild#get-guild-roles
```
Comment (UndarkAido) [inline discordpp/categories/guild.hh]: I'd like to try to keep stuff in the same order as the API so this goes after the last 2 endpoints

File: discordpp/categories/guild.hh [hunk 2]
```diff
@@ -492,3 +491,79 @@ sptr<const std::string> render_target() override {
...
+    HIDE_FIELD(body)                                                           \
+    FORWARD_FIELD(handleWrite, onWrite, )                                      \
+    FORWARD_FIELD(handleRead, onRead, )
+
+#include <discordpp/macros/defineCallOpen.hh>
+protected:
+sptr<const std::string> render_target() override {
+    std::string out = fmt::format("/guilds/{}/members/search", *_guild_id);
+    bool first = true;
+    if (!_guild_id) {
```
Comment (UndarkAido) [inline discordpp/categories/guild.hh:489]: The `_guild_id` check should come before `_guild_id` is used to make `out`

File: discordpp/categories/guild.hh [hunk 3]
```diff
@@ -492,3 +491,79 @@ sptr<const std::string> render_target() override {
...
+        throw std::logic_error("Search Guild Members needs a Guild ID");
+    }
+    if(!_query) {
+        throw std::logic_error("Search Guild Members needs a Query");
+    } else {
+        out += fmt::format("{}query={}", first ? "?" : "&", *_query);
+        first = false;
```
Comment (UndarkAido) [inline discordpp/categories/guild.hh]: You don't need `first` if `query` is required- just put the check before `out` and include it in that

File: discordpp/categories/guild.hh [hunk 4]
```diff
@@ -492,3 +491,79 @@ sptr<const std::string> render_target() override {
...
+    HIDE_FIELD(body)                                                           \
+    HIDE_FIELD(payload)                                                        \
+    FORWARD_FIELD(handleWrite, onWrite, )                                      \
+    FORWARD_FIELD(handleRead, onRead, )
+
+#include <discordpp/macros/defineCallOpen.hh>
+protected:
+sptr<const std::string> render_target() override {
+    if (!_guild_id) {
+        throw std::logic_error("Search Guild Members needs a Guild ID");
```
Comment (UndarkAido) [inline discordpp/categories/guild.hh]: Wrong error string

File: discordpp/categories/guild.hh [hunk 5]
```diff
@@ -492,3 +491,79 @@ sptr<const std::string> render_target() override {
...
+    }
+    if(!_query) {
+        throw std::logic_error("Search Guild Members needs a Query");
+    } else {
+        out += fmt::format("{}query={}", first ? "?" : "&", *_query);
+        first = false;
+    }
+    if (_limit) {
+        out += fmt::format("{}limit={}", first ? "?" : "&", *_limit);
```
Comment (UndarkAido) [inline discordpp/categories/guild.hh]: And as such this `out +=` doesn't need the ternary

File: discordpp/categories/guild.hh [hunk 6]
```diff
@@ -492,3 +491,79 @@ sptr<const std::string> render_target() override {
...
+    }
+    return std::make_shared<const std::string>(
+        fmt::format("/guilds/{}/members/@me/nick", *_guild_id));
+}
+sptr<const json> render_payload() override {
+    json out;
+    if(_nick) {
+        out["nick"] = *_nick;
+    } else {
+        out["nick"] = "";
```
Comment (UndarkAido) [inline discordpp/categories/guild.hh:534]: Could you test what happens when the field is empty vs. missing? If so it should be a `std::optional` with a test for that too.

**Review by UndarkAido — CHANGES_REQUESTED** [review]
Wow, thank you for adding more of these! Just a couple comments this time- nothing major. And I'm sure you are, but I wanted to check that when you mark something as verified you're testing all of the parameters?

File: discordpp/categories/guild.hh (hunk 6 above)
Comment (noahknegt) [reply discordpp/categories/guild.hh:534]: if its missing it does not do anything that is why i added the the empty quotes

File: discordpp/categories/guild.hh (hunk 1 above)
Comment (noahknegt) [reply discordpp/categories/guild.hh]: I will swap them around then, sorry for that

File: discordpp/categories/guild.hh [hunk 7]
```diff
@@ -512,20 +486,17 @@ sptr<const std::string> render_target() override {
 #include <discordpp/macros/defineCallOpen.hh>
 protected:
 sptr<const std::string> render_target() override {
-    std::string out = fmt::format("/guilds/{}/members/search", *_guild_id);
-    bool first = true;
     if (!_guild_id) {
         throw std::logic_error("Search Guild Members needs a Guild ID");
     }
+    std::string out = fmt::format("/guilds/{}/members/search", *_guild_id);
     if(!_query) {
```
Comment (UndarkAido) [inline discordpp/categories/guild.hh:492]: Put this before `out`

File: discordpp/categories/guild.hh [hunk 8]
```diff
@@ -512,20 +486,17 @@ sptr<const std::string> render_target() override {
...
         throw std::logic_error("Search Guild Members needs a Query");
     } else {
-        out += fmt::format("{}query={}", first ? "?" : "&", *_query);
-        first = false;
+        out += fmt::format("?query={}", *_query);
```
Comment (UndarkAido) [inline discordpp/categories/guild.hh]: Add this to `out`'s declaration

**Review by UndarkAido — CHANGES_REQUESTED** [review]
Just this one last thing, sorry if I wasn't clear

File: discordpp/categories/guild.hh (hunk 6 above)
Comment (UndarkAido) [reply discordpp/categories/guild.hh:534]: Even if the user has a nickname already? Weird. I wonder why they'd include that.

File: discordpp/categories/guild.hh (hunk 8 above)
Comment (noahknegt) [reply discordpp/categories/guild.hh]: so like this "std::string out = fmt::format("/guilds/{}/members/search?query={}", *_guild_id, *_query);"?

Comment (UndarkAido) [reply discordpp/categories/guild.hh]: Yes

You can use backticks for inline code fyi: `

**Pull request merged**